enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,294 @@
+* 4.15  GA as opt-in
+* 4.16 GA as opt-in, warn about prometheus-adapter being deprecated if `prometheus-adapter` config isn’t empty and `metrics-server` isn’t already enabled.
+* 4.17 GA, enabled by default
+* 4.18 => Remove `prometheus-adapter` from CMO
+
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+More testing (upgrade, downgrade, scale).
+* Sufficient time for feedback including signals from telemetry about the customer adoption (e.g. number of metrics-server deployment across the fleet).
+* Counter-measures to avoid service degradation of the Kubelet.
+* Conduct load testing.
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+### Version Skew Strategy
+
+TBD
+
+### Operational Aspects of API Extensions
+
+`metrics-server` will be deployed in HA mode (2 replicas by default). We will define Pod AntiAffinity Rule in `metrics-server` Deployment spec and create `PodDisruptionBudget` config to meet HA criteria.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: metrics-server
+    app.kubernetes.io/name: metrics-server
+    app.kubernetes.io/part-of: openshift-monitoring
+  name: metrics-server
+  namespace: openshift-monitoring
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: metrics-server
+      app.kubernetes.io/component: metrics-server
+      app.kubernetes.io/part-of: openshift-monitoring
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 1
+  template:
+    metadata:
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+      labels:
+        app.kubernetes.io/component: metrics-server
+        app.kubernetes.io/name: metrics-server
+        app.kubernetes.io/part-of: openshift-monitoring
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: metrics-server
+                app.kubernetes.io/name: metrics-server
+                app.kubernetes.io/part-of: openshift-monitoring
+            namespaces:
+            - openshift-monitoring
+            topologyKey: kubernetes.io/hostname
+...
+```
+
+```yaml
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  labels:
+    app.kubernetes.io/name: metrics-server
+    app.kubernetes.io/part-of: openshift-monitoring
+  name: metrics-server
+  namespace: openshift-monitoring
+spec:
+  minAvailable: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: metrics-server
+      app.kubernetes.io/component: metrics-server
+      app.kubernetes.io/part-of: openshift-monitoring
+```
+
+#### Failure Modes
+
+TBD
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+N/A
+
+## Alternatives
```

Comment:
We also discussed using a [config] CRD to configure metrics-server. This might be worth mentioning here as well.